lib/pages/settings_list_item.dart

```diff
@@ -76,16 +76,23 @@ class SlowMotionSetting extends StatelessWidget {
 class SettingsListItem<T> extends StatefulWidget {
   SettingsListItem({
     Key key,
+    @required this.optionsMap,
     @required this.title,
-    @required this.options,
     @required this.selectedOption,
     @required this.onOptionChanged,
     @required this.onTapSetting,
     @required this.isExpanded,
-  }) : super(key: key);
+  })  : options = optionsMap.keys,
+        displayOptions = optionsMap.values,
+        super(key: key);
+
+  final LinkedHashMap<T, DisplayOption> optionsMap;
+
+  /// For ease of use. Correspond to the keys and values of [optionsMap].
```

Comment:
I feel like breaking the map into these 2 Iterables doesn't add much value. I suggest removing them.

If you think they should remain, then please make them private.

Reply:
Doing data structure operations felt kind of wrong in the build method, although that's purely based on feeling. I'll see to moving this to the _State class

Reply:
Did a bit of reading on these things and because it's implemented as a hash map + doubly linked list, obtaining the keys/values is inexpensive.